### DISCUSSION TITLE
Children nodes as automatic ordered list

### DISCUSSION THREAD

**Discussion by euu2021** [freeplane/freeplane#1364] [Show and tell]
Sometimes I would like to have the children nodes working like an ordered list in a word processor. Today, I stopped to think a bit about it, and found an easy solution using conditional styles:

1.  Create a new style, that has the Node numbering activated
![image](https://github.com/freeplane/freeplane/assets/77707706/a5064ee7-b387-44ad-ab5c-aa87dea07a8e)

2. Create a conditional style, like this:
![image](https://github.com/freeplane/freeplane/assets/77707706/31a069eb-2eb5-4147-b7d8-668ce8da82b0)

3. Create the attribute "list" to the node that will be the parent of the ordered list:
![image](https://github.com/freeplane/freeplane/assets/77707706/ce8be938-1e72-4ca6-9aa2-c75ad104108d)

I used an attribute as the trigger for the conditional style, but it could be an icon, a style, or a piece of text in the core, for example.

**Comment by sifran-github** [comment]
i'm not sure what you are trying to achieve with the conditional styles.

You can have a similar effect if you select the children and then mark their core text to be "Node numbering" (like your first step), without any special style. and if you want that additional sibling nodes will get the same style with the node numbering, you can select: format -> copy format to new sibling nodes.

I mean, this is the simplest solution that i could think about. i'm just not sure what you are trying to accomplish. maybe your way gives you some additional functionality that i have missed.

**Reply by euu2021** [reply]
> You can have a similar effect if you select the children and then mark their core text to be "Node numbering" (like your first step), without any special style. and if you want that additional sibling nodes will get the same style with the node numbering, you can select: format - copy format to new sibling nodes.

This works if the user keeps the `format -> copy format to new sibling nodes` activated. Otherwise, he needs to keep activating and deactivating this setting every time he works on the list, which is not practical.

Some time ago, I tried `format -> copy format to new sibling nodes`, but it did not fit well in my workflow, for some reason. I will give it a try, again, to see if I gave up too soon.

**Reply by sifran-github** [reply]
yeah, in the end its a personal choice.

I always use format -> copy format to new sibling nodes, and I almost never configure complex styles with conditionals. If I want to create a new default node, I click on the parent and insert a new child node, and if I want the same node type, I start from a node and create a sibling.

but its good that Freeplane allows many ways to configure and personalize it.

**Reply by euu2021** [reply]
> You can have a similar effect if you select the children and then mark their core text to be "Node numbering" (like your first step), without any special style. and if you want that additional sibling nodes will get the same style with the node numbering, you can select: format - copy format to new sibling nodes.

Testing a bit more I see that the `format -> copy format to new sibling nodes` solution has some limitations, because the numbering is not automatically applied when pasting nodes, drag and dropping nodes, or creating nodes from the parent, for example. The conditional style deals with that automatically.

**Reply by sifran-github** [reply]
I see. yes, in case you need this functionality, then probably having a conditional style is a good option.

I have another suggestion:
if you want to support multiple levels of children (just like a word processor when you have a list and use tab for indentation), you can change the filter condition "parent" to "Any ancestor". This way, any child under the list node (including children of children and so on) will be numbered.

or you can separate it to different filters such as "list" and "list_all"

**Reply by euu2021** [reply]
> I have another suggestion:
if you want to support multiple levels of children (just like a word processor when you have a list and use tab for indentation), you can change the filter condition "parent" to "Any ancestor". This way, any child under the list node (including children of children and so on) will be numbered.
or you can separate it to different filters such as "list" and "list_all"

This is nice. It gets even more similar to the behavior in a word processor
